include/lo2s/perf/counter/counter_collection.hpp

```diff
@@ -33,18 +33,18 @@ namespace counter
 {
 struct CounterCollection
 {
-    EventDescription leader;
-    std::vector<EventDescription> counters;
+    SysfsEvent leader;
```

Comment:
I've grouped them into SysfsEvent, because they all used the EventDescription struct before and have the same sample_type.
I'd like to keep them like this (maybe with a different name) or, if that isn't an option, create a new class for group / userspace counter- and leader-events. But putting this into the base class PerfEvent would give all other event classes unnecessary functions that they don't use.

Reply:
From my pov it looked like it was using the sysfsevent code for everything, even things not necessarily coming from sysfs. If it is not strictly sysfs but also not the perfevent base class, then introduce another class.